modules/seqexec/server/src/main/scala/seqexec/server/nifs/Nifs.scala

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server.nifs
+
+import cats.data.Reader
+import cats.effect.Sync
+import cats.effect.IO
+import cats.effect.LiftIO
+import cats.implicits._
+import edu.gemini.spModel.config2.Config
+import edu.gemini.spModel.gemini.nifs.InstNIFS._
+import edu.gemini.spModel.gemini.nifs.InstEngNifs._
+import gem.enum.LightSinkName
+import java.lang.{ Double => JDouble }
+import java.lang.{ Integer => JInt }
+import java.beans.PropertyDescriptor
+import shapeless.tag
+import shapeless.tag.@@
+import seqexec.server.ConfigUtilOps._
+import seqexec.model.dhs.ImageFileId
+import seqexec.model.enum.Instrument
+import seqexec.model.enum.Resource
+import seqexec.server.ConfigUtilOps.ExtractFailure
+import seqexec.server.ConfigResult
+import seqexec.server.InstrumentSystem
+import seqexec.server.ObserveCommand
+import seqexec.server.Progress
+import seqexec.server.SeqActionF
+import seqexec.server.SeqObserveF
+import seqexec.server.TrySeq
+import seqexec.server.keywords.DhsClient
+import seqexec.server.keywords.DhsInstrument
+import seqexec.server.keywords.KeywordsClient
+import seqexec.server.InstrumentSystem.AbortObserveCmd
+import seqexec.server.InstrumentSystem.InfraredControl
+import seqexec.server.InstrumentSystem.StopObserveCmd
+import seqexec.server.nifs.NifsController._
+import squants.Time
+import squants.time.TimeConversions._
+
+final case class Nifs[F[_]: LiftIO: Sync](controller: NifsController[IO], // TODO Convert to NifsController[F]
+                                          dhsClient:  DhsClient[F])
+    extends DhsInstrument[F]
+    with InstrumentSystem[F] {
+
+  import Nifs._
+
+  override def sfName(config: Config): LightSinkName = LightSinkName.Nifs
+
+  override val contributorName: String = "NIFS"
+
+  override val observeControl: InstrumentSystem.ObserveControl = {
+    // Ideally controller.stopObserver should produce F[Unit] rather than IO[Unit]
+    // but that opens a can of worms
+    InfraredControl(StopObserveCmd(SeqActionF.liftF(controller.stopObserve)),
+                    AbortObserveCmd(SeqActionF.liftF(controller.abortObserve)))
+
+  }
+
+  override def observe(
+    config: Config
+  ): SeqObserveF[F, ImageFileId, ObserveCommand.Result] =
+    Reader { fileId =>
+      SeqActionF
+        .either(getDCConfig(config).asTrySeq)
+        .flatMap(x => SeqActionF.liftIO(controller.observe(fileId, x)))
+    }
+
+  override def calcObserveTime(config: Config): Time =
+    getDCConfig(config)
+      .map(controller.calcTotalExposureTime)
+      .getOrElse(60.seconds)
```

Comment:
Seems like a magic number. Is this not available / defined elsewhere?

Reply:
This is a good comment. @jluhrs After looking around shouldn't we fail if we cannot calculate the exp time?

Reply:
So this should likely be improved but it affects each instrument. I think we should return `Option[Time]`